istioctl/cmd/add-to-mesh.go

```diff
@@ -0,0 +1,177 @@
+	addToMeshCmd.PersistentFlags().StringVar(&injectConfigMapName, "injectConfigMapName", defaultInjectConfigMapName,
+		fmt.Sprintf("ConfigMap name for Istio sidecar injection, key should be %q.", injectConfigMapKey))
+
+	return addToMeshCmd
+}
+
+func svcMeshifyCmd() *cobra.Command {
+	cmd := &cobra.Command{
+		Use:   "service",
+		Short: "Add Service into Istio Service Mesh",
+		Long: `Find deployments for the service, inject side car and add the service into mesh""
+
+THIS COMMAND IS STILL UNDER ACTIVE DEVELOPMENT AND NOT READY FOR PRODUCTION USE.
+`,
+		Example: `istioctl experimental add-to-mesh service productpage`,
+		RunE: func(cmd *cobra.Command, args []string) error {
+			if len(args) != 1 {
+				return fmt.Errorf("expecting service name")
+			}
+			client, err := interfaceFactory(kubeconfig)
+			if err != nil {
+				return err
+			}
+			ns := handlers.HandleNamespace(namespace, defaultNamespace)
+
+			var meshConfig *meshconfig.MeshConfig
+			writer := cmd.OutOrStdout()
+			if meshConfigFile != "" {
+				if meshConfig, err = istiocmd.ReadMeshConfig(meshConfigFile); err != nil {
+					return err
+				}
+			} else {
+				if meshConfig, err = getMeshConfigFromConfigMap(kubeconfig); err != nil {
+					return err
+				}
+			}
+
+			var sidecarTemplate string
+			if injectConfigFile != "" {
+				injectionConfig, err := ioutil.ReadFile(injectConfigFile) // nolint: vetshadow
+				if err != nil {
+					return err
+				}
+				var injectConfig inject.Config
+				if err := yaml.Unmarshal(injectionConfig, &injectConfig); err != nil {
+					return multierr.Append(fmt.Errorf("loading --injectConfigFile"), err)
+				}
+				sidecarTemplate = injectConfig.Template
+			} else if sidecarTemplate, err = getInjectConfigFromConfigMap(kubeconfig); err != nil {
+				return err
+			}
+
+			var valuesConfig string
+			if valuesFile != "" {
+				valuesConfigBytes, err := ioutil.ReadFile(valuesFile) // nolint: vetshadow
+				if err != nil {
+					return err
+				}
+				valuesConfig = string(valuesConfigBytes)
+			} else if valuesConfig, err = getValuesFromConfigMap(kubeconfig); err != nil {
+				return err
+			}
+
+			svc, err := client.CoreV1().Services(ns).Get(args[0], metav1.GetOptions{})
+			if err != nil {
+				return err
+			}
+			svcSelector := k8s_labels.SelectorFromSet(svc.Spec.Selector)
+			matchingDeployments := []appsv1.Deployment{}
+			deployments, err := client.AppsV1().Deployments(ns).List(metav1.ListOptions{})
+			if err != nil {
+				return err
+			}
+			for _, dep := range deployments.Items {
+				depLabels := k8s_labels.Set(dep.ObjectMeta.Labels)
+				if svcSelector.Matches(depLabels) {
+					matchingDeployments = append(matchingDeployments, dep)
+				}
+			}
+			if len(matchingDeployments) == 0 {
+				fmt.Fprintf(writer, "No deployments found for service %s.%s", svc.Name, svc.Namespace)
+				return nil
+			}
+			for _, dep := range matchingDeployments {
```

Comment:
Would it be better to use `multierror` and not return until we've finished attempting all of them?

Reply:
thanks for the suggestion! fixed